notebooks/A_CNN_Annotation_with_Napari.ipynb

```diff
@@ -197,39 +206,42 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 6,
    "metadata": {},
    "outputs": [],
    "source": [
     "def get_image_patch(layers, coords, shape=64):\n",
     "    \"\"\" Get an image patch from the image layer data. \"\"\"\n",
     "    \n",
-    "    frame, y_coo, x_coo = [int(coo) for coo in coords]\n",
-    "    pad = shape // 2\n",
-    "    \n",
-    "    # Read the image from layers:\n",
-    "    # TODO: soft-code ## presuming this has already been done? nd.\n",
-    "    image = layers[0].data[frame]\n",
-    "    im_h, im_w = image.shape\n",
-    "    edge_dist = [y_coo, im_h - y_coo, x_coo, im_w - x_coo]\n",
-    "    padded = \"\"\n",
-    "    \n",
-    "    # Check if padding is needed:\n",
-    "    if any([item < pad for item in edge_dist]):\n",
-    "        pad_width = int(pad - min(edge_dist))\n",
-    "        image = np.pad(array=image, pad_width=pad_width, mode='constant')\n",
-    "        y_coo += pad_width\n",
-    "        x_coo += pad_width\n",
-    "        padded = \"_flagged\"\n",
-    "    \n",
-    "    # Crop patch around the point:\n",
-    "    patch = np.array([row[x_coo - pad : x_coo + pad] for row in image[y_coo - pad : y_coo + pad]]).astype(np.uint8)\n",
+    "    flagged = False\n",
+    "    square = shape // 2\n",
+    "\n",
+    "    for layer in layers:\n",
+    "        \n",
+    "        # Read the image from layers:\n",
+    "        frame, y_coo, x_coo = [int(coo) for coo in coords]\n",
+    "        image = layer.data[frame]\n",
+    "        im_h, im_w = image.shape\n",
+    "        edge_dist = [y_coo, im_h - y_coo, x_coo, im_w - x_coo]\n",
+    "        \n",
+    "        # Check if padding is needed:\n",
+    "        if any([item < square for item in edge_dist]):\n",
+    "            pad_width = int(square - min(edge_dist))\n",
+    "            image = np.pad(array=image, pad_width=pad_width, mode='constant')\n",
+    "            y_coo = y_coo + pad_width\n",
+    "            x_coo = x_coo + pad_width\n",
+    "            flagged = True\n",
+    "\n",
+    "        # Slice the patch from the image:\n",
+    "        patches = image[y_coo-square : y_coo+square, x_coo-square : x_coo+square].astype(np.uint8)\n",
+    "        patches = np.stack(patches, axis=-1)\n",
     "    \n",
-    "    # Check if patches are of specified shape:\n",
-    "    if patch.shape != (shape, shape):\n",
-    "        raise ValueError(f\"Image patch doesn't have correct shape: {patch.shape}\")\n",
+    "    # Check if all patches are of specified shape:\n",
+    "    #if not all([patch.shape == (shape, shape) for patch in patches[0]]):\n",
+    "    #    raise ValueError(f\"Image patches don't have correct shape: {[patch.shape for patch in patches[0]]}\")\n",
     "    \n",
-    "    return [patch, padded]\n"
+    "    return patches, flagged\n",
```

Comment:
I've edited the code according the instructions now. Seems to work pretty well.

Reply:
Just a detail, if the output isn't useful for the user, might be good to go to the "Kernel" tab and choose "Restart & Clear Output" to reset the notebook to a clean state.